test/integration/puppetlabs/services/legacy_routes/legacy_routes_test.clj

```diff
@@ -50,6 +54,75 @@
        (is (= "localhost" (get cert-status-body "name")))
        (is (= "signed" (get cert-status-body "state")))))))
 
+(deftest ^:integration legacy-routes-metrics
+  (bootstrap/with-puppetserver-running
+   app
+   {}
+   (let [jruby-metrics-service (tk-app/get-service app :JRubyMetricsService)
+         svc-context (tk-services/service-context jruby-metrics-service)
+         jruby-metrics (:metrics svc-context)
+         jruby-service (tk-app/get-service app :JRubyPuppetService)
+         time-before-first-borrow (System/currentTimeMillis)]
+     ;; Use with-jruby-puppet to borrow the lone jruby out of the pool,
+     ;; which should cause a subsequent catalog request to block on a borrow
+     ;; request
+     (jruby-service/with-jruby-puppet
+      _
+      jruby-service
+      :legacy-routes-test
+      (let [time-before-second-borrow (System/currentTimeMillis)]
+        (future
+         (logutils/with-test-logging
+          (http-get "/puppet/v3/catalog/localhost?environment=production")))
```

Comment:
d'oh.  Copypasta problem, I think.  I changed this to actually use a legacy route in 20e9ccc.